docs/operations/upgrade.md

```diff
@@ -0,0 +1,51 @@
+# Upgrading Loki
+
+Every attempt is made to keep Loki backwards compatible, such that upgrades should be low risk and low friction.
+
+Unfortunately Loki is software and software is hard and sometimes things are not as easy as we want them to be.
+
+On this page we will document any upgrade issues/gotchas/considerations we are aware of.
+
+## 1.4.0
+
+Loki 1.4.0 vendors Cortex v0.7.0-rc.0 which contains [several breaking config changes](https://github.com/cortexproject/cortex/blob/v0.7.0-rc.0/CHANGELOG.md), however, you would only be affected if you were configuring your schema via arguments and not a config file.  This is not something we had ever provided as an option via docs and is unlikely anyone is doing, but worth mentioning.  None of the other config changes are relevant to Loki.
+
+### Required Upgrade Path
+
+The newly vendored version of Cortex removes code related to de-normalized tokens in the ring. What you need to know is this:
+
+*Note:* A "shared ring" as mentioned below refers to using *consul* or *etcd* for values in the following config:
+
+```yaml
+kvstore:
+  # The backend storage to use for the ring. Supported values are
+  # consul, etcd, inmemory
+  store: <string>
+```
+
+* Running without using a shared ring (inmemory): No action required
+* Running with a shared ring and upgrading from v1.3.0 -> v1.4.0: No action required
+* Running with a shared ring and upgrading from any version less than v1.3.0 (e.g. v1.2.0) -> v1.4.0: **ACTION REQUIRED**
+
+There are two options for upgrade if you are not on version 1.3.0 and are using a shared ring:
+
+* Upgrade first to v1.3.0 **BEFORE** upgrading to v1.4.0
+
+OR 
+
+**Note:** If you are running a single binary you only need to add this flag to your single binary command.
+
+1. Add the following configuration to your ingesters command: `-ingester.normalise-tokens=true`
+1. Restart your ingesters with this config
+1. Proceed with upgrading to v1.4.0
+
+**Note:** It's also possible to enable this flag via config file, see the [docs](https://github.com/grafana/loki/tree/v1.3.0/docs/configuration#lifecycler_config)
```

Comment:
For helm you can use add this to you value file if you're using loki

```
extraArgs:
  ingester.normalise-tokens: true
```

or if you're using loki-stack:
```
loki:
  extraArgs:
    ingester.normalise-tokens: true